lib/_http_outgoing.js

```diff
@@ -74,13 +72,63 @@ function OutgoingMessage() {
   this.socket = null;
   this.connection = null;
   this._header = null;
-  this._headers = null;
+  this[outHeadersKey] = null;
 
   this._onPendingData = null;
 }
 util.inherits(OutgoingMessage, Stream);
 
 
+Object.defineProperty(OutgoingMessage.prototype, '_headers', {
+  get: internalUtil.deprecate(function() {
+    return this.getHeaders();
+  }, 'msg._headers is deprecated. Use public API methods instead.'),
+  set: internalUtil.deprecate(function(val) {
+    if (val == null) {
+      this[outHeadersKey] = null;
+    } else if (typeof val === 'object') {
```

Comment:
What odd effects? Prior to all of these http changes, unless there were numeric headers set in `._headers`, there would never be a match when node would do `._headerNames[lowercasedName]` (which would have already been causing issues on its own for quite awhile now I would assume) to find the original-cased name when rendering the headers.